docs/how-to/cooperative_groups.rst

```diff
@@ -0,0 +1,412 @@
+
+The ``thread_rank()`` , ``size()``, ``cg_type()``, ``is_valid()``, ``sync()``, 
+``meta_group_rank()``, ``meta_group_size()``, ``shfl(...)``, ``shfl_down(...)``, 
+``shfl_up(...)`` and ``shfl_xor(...)`` member functions are public of the ``thread_block_tile``
+class. For further details check the :ref:`thread_block_tile references <thread_block_tile_ref>` . 
+
+Coalesced Groups
+------------------
+
+Represents an active thread group in a wavefront. This group type also supports sub-wave level
+intrinsics.
+
+.. code-block:: C++
+
+  class coalesced_group;
+
+Constructed via:
+
+.. code-block:: C++
+
+  coalesced_group active = coalesced_threads();
+
+.. note::
+  
+  * ``shfl`` functions are currently supports integer or float type.
+
+The ``thread_rank()`` , ``size()``, ``cg_type()``, ``is_valid()``, ``sync()``, 
+``meta_group_rank()``, ``meta_group_size()``, ``shfl(...)``, ``shfl_down(...)``, 
+and ``shfl_up(...)`` member functions are public of the ``coalesced_group`` class. 
+For further details check the :ref:`coalesced_group references <coalesced_group_ref>` .
+
+Synchronization
+=================
+
+At different types of groups, the synchronization requires using the correct cooperative group
+launch API.
+
+Thread-Block Synchronization
+-----------------------------------------------
+
+1. The new block representation can be accessed with the original kernel launch methods.
+2. The device side synchronization is written in the following form.
+
+.. code-block:: C++
+
+  thread_block g = this_thread_block();
+  g.sync();
+
+Grid Synchronization
+---------------------
+
+This section describes the necessary step to be able to synchronize group over a single GPU:
+
+1. Check the cooperative launch capability on single AMD GPU:
+
+.. code-block:: C++
+
+    int device               = 0;
+    int supports_coop_launch = 0;
+    // Check support
+    // Use hipDeviceAttributeCooperativeMultiDeviceLaunch when launching across multiple devices
+    HIP_CHECK(hipGetDevice(&device));
+    HIP_CHECK(
+        hipDeviceGetAttribute(&supports_coop_launch, hipDeviceAttributeCooperativeLaunch, device));
+    if(!supports_coop_launch)
+    {
+        std::cout << "Skipping, device " << device << " does not support cooperative groups"
+                  << std::endl;
+        return 0;
+    }
+
+2. Launch the cooperative kernel on single GPU:
+
+.. code-block:: C++
+
+    void* params[] = {&d_vector, &d_block_reduced, &d_partition_reduced};
+    // Launching kernel from host.
+    HIP_CHECK(hipLaunchCooperativeKernel(vector_reduce_kernel<partition_size>,
+                                         dim3(num_blocks),
+                                         dim3(threads_per_block),
+                                         params,
+                                         0,
+                                         hipStreamDefault));
+
+
+3. The device side synchronization on single GPU:
```

Comment:
Don't understand this - what is the procedure

Reply:
This is actual synchronization of the threads over the grid. This section describes the necessary steps to be able to synchronize with the different types of groups.